lib/Mojolicious.pm

```diff
@@ -35,7 +35,7 @@ has log              => sub {
 };
 has 'max_request_size';
 has mode               => sub { $ENV{MOJO_MODE} || $ENV{PLACK_ENV} || 'development' };
-has moniker            => sub { Mojo::Util::decamelize ref shift };
+has moniker            => sub { Mojo::Util::class_to_file ref shift };
```

Comment:
This is a breaking change unfortunately. This has implications about how configuration files are loaded by default. I'd be fine changing this at a major version or at least in line with our breaking changes rules (though I would personally rather this be at a major version)

Reply:
Is there a process I should follow to put this forward for a major release?

Reply:
While I support your stance @jberger that this is a breaking change and thus should be planned for a major release, it is also a question of definition: is the proposed change a _change_ or a _bugfix_?

When looking over a couple of projects here I see that we usually end up overriding `moniker` because of how the default moniker is generated (see example table from the PR description). So from that perspective if this change was to be implemented we no longer would have to set `moniker` in our apps, because the problem with the auto-generated moniker would go away.

From that perspective I think it is fair to view this as a bugfix, and thus be considered for a minor-release.